client/task_runner.go

```diff
@@ -44,17 +47,22 @@ type TaskStateUpdater func(taskName, status, desc string)
 // NewTaskRunner is used to create a new task context
 func NewTaskRunner(logger *log.Logger, config *config.Config,
 	updater TaskStateUpdater, ctx *driver.ExecContext,
-	allocID string, task *structs.Task) *TaskRunner {
+	allocID string, task *structs.Task, taskType string,
+	restartPolicy *structs.RestartPolicy) *TaskRunner {
+
+	rt := newRestartTracker(taskType, restartPolicy)
```

Comment:
Lets just pass the RestartTracker in, and then we don't need the Policy or task type